src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,145 @@
+package duke;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The Parser class, containing the parsing logic for the bot.
+ *
+ * @author Jet Tan
+ */
+public class Parser {
+    /**
+     * Processes the input.
+     *
+     * @param input input of the user
+     * @throws DukeException duke.InvalidInputException, EmptyDescDescription, UnknownCommandException
+     */
+    public static void process(String input) throws DukeException, IOException {
+        String[] arr = input.split(" ");
+        String command = arr[0]; // first word of the user input
+        switch (command) {
+        case "bye":
```

Comment:
Switch statement indentation is correctly followed according to module guidelines!! 😄